sdks/java/core/src/test/java/org/apache/beam/sdk/util/common/CounterTest.java

```diff
@@ -588,4 +588,148 @@ public void testMergeIncompatibleCounters() {
+    assertEquals(DirtyBit.COMMITTED, doubleSum.dirty.get());
+    assertEquals(DirtyBit.COMMITTED, doubleMean.dirty.get());
+    assertEquals(DirtyBit.COMMITTED, intSum.dirty.get());
+    assertEquals(DirtyBit.COMMITTED, intMean.dirty.get());
+    assertEquals(DirtyBit.COMMITTED, boolAnd.dirty.get());
+
+    // Test counters are dirty after mutating.
+    longSum.addValue(1L);
+    longMean.resetMeanToValue(1L, 1L);
+    doubleSum.addValue(1.0);
+    doubleMean.resetMeanToValue(1L, 1.0);
+    intSum.addValue(1);
+    intMean.resetMeanToValue(1, 1);
+    boolAnd.addValue(true);
+
+    assertTrue(longSum.isDirty());
+    assertTrue(longMean.isDirty());
+    assertTrue(doubleSum.isDirty());
+    assertTrue(doubleMean.isDirty());
+    assertTrue(intSum.isDirty());
+    assertTrue(intMean.isDirty());
+    assertTrue(boolAnd.isDirty());
+
+    assertEquals(DirtyBit.DIRTY, longSum.dirty.get());
+    assertEquals(DirtyBit.DIRTY, longMean.dirty.get());
+    assertEquals(DirtyBit.DIRTY, doubleSum.dirty.get());
+    assertEquals(DirtyBit.DIRTY, doubleMean.dirty.get());
+    assertEquals(DirtyBit.DIRTY, intSum.dirty.get());
+    assertEquals(DirtyBit.DIRTY, intMean.dirty.get());
+    assertEquals(DirtyBit.DIRTY, boolAnd.dirty.get());
+
+    // Test counters are dirty and are COMMITTING.
+    longSum.committing();
+    longMean.committing();
+    doubleSum.committing();
+    doubleMean.committing();
+    intSum.committing();
+    intMean.committing();
+    boolAnd.committing();
+
+    assertTrue(longSum.isDirty());
+    assertTrue(longMean.isDirty());
+    assertTrue(doubleSum.isDirty());
+    assertTrue(doubleMean.isDirty());
+    assertTrue(intSum.isDirty());
+    assertTrue(intMean.isDirty());
+    assertTrue(boolAnd.isDirty());
+
+    assertEquals(DirtyBit.COMMITTING, longSum.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, longMean.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, doubleSum.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, doubleMean.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, intSum.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, intMean.dirty.get());
+    assertEquals(DirtyBit.COMMITTING, boolAnd.dirty.get());
+
+    // Test counters are dirty again after mutating.
+    longSum.addValue(1L);
+    longMean.resetMeanToValue(1L, 1L);
+    doubleSum.addValue(1.0);
+    doubleMean.resetMeanToValue(1L, 1.0);
+    intSum.addValue(1);
+    intMean.resetMeanToValue(1, 1);
+    boolAnd.addValue(true);
+
+    longSum.committed();
+    longMean.committed();
+    doubleSum.committed();
+    doubleMean.committed();
+    intSum.committed();
+    intMean.committed();
+    boolAnd.committed();
+
+    assertTrue(longSum.isDirty());
+    assertTrue(longMean.isDirty());
+    assertTrue(doubleSum.isDirty());
+    assertTrue(doubleMean.isDirty());
+    assertTrue(intSum.isDirty());
+    assertTrue(intMean.isDirty());
+    assertTrue(boolAnd.isDirty());
+
+    assertEquals(DirtyBit.DIRTY, longSum.dirty.get());
```

Comment:
If you `assertTrue(counter.committing())` instead of checking the explicit value of the DirtyBit you can just demonstrate the API behaves appropriately